src/Components/Compute/AuctionBlock/Auction.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+import AuctionBlock from '../../UI/AuctionBlock/AuctionBlock';
+
+export default class Auction extends React.Component {
+  constructor(props) {
+    super(props);
+    this.initialState = {
+      biddingSequence: props.biddingSequence,
+      highestBid: null,
+      winningPlayer: null,
+      currentBid: null,
+    };
+    this.state = { ...this.initialState };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { biddingSequence, name } = this.props;
+    if (prevProps.name !== name) {
+      this.setState({ biddingSequence });
+    }
+  }
+
+  onBidSubmit = () => {
+    let { currentBid, highestBid, biddingSequence, winningPlayer } = this.state;
+    if (!highestBid || currentBid > highestBid) {
+      highestBid = currentBid;
+      winningPlayer = biddingSequence[0];
+      biddingSequence = [...biddingSequence.slice(1), biddingSequence[0]];
+      return this.setState({ highestBid, biddingSequence, winningPlayer });
+    }
+    biddingSequence = [...biddingSequence.slice(1), biddingSequence[0]];
+    return this.setState({ biddingSequence });
```

Comment:
Why returning `this.stateState({biddingSequence})` ?
Are you using returned value somewhere?